components/sp-solutions/org.wso2.analytics.solutions.sp.http.analytics/widgets/HTTPAnalyticsRequestCountFilter/src/HTTPAnalyticsRequestCountFilter.jsx

```diff
@@ -270,17 +425,45 @@ class HTTPAnalyticsRequestCountFilter extends Widget {
             disabled: false,
         }));
 
+        const userSelection = this.getInitialUserSelection(servers, services);
         this.setState({
             servers,
             services,
             serviceOptions,
             serverOptions,
-            selectedServerValues: allOption,
-            selectedServiceValues: allOption,
-            selectedSingleServiceValue: allOption[0],
+            selectedServerValues: userSelection.servers,
+            selectedServiceValues: userSelection.services,
+            selectedSingleServiceValue: userSelection.method,
         }, this.publishUpdate);
     }
 
+    /**
+     * Sort in ascending order
+     * @param a
+     * @param b
+     */
+    naturalSort(a, b) {
```

Comment:
Change as suggested in code review

Reply:
This is the comparator function passed to the sort method, not a sort function.